Tasks/UseDotNetV2/Tests/globaljsonfetcherTest.ts

```diff
@@ -0,0 +1,95 @@
+"use strict";
+import * as tl from 'azure-pipelines-task-lib/task';
+import { GlobalJson } from "../globaljsonfetcher";
+import { Buffer } from "buffer";
+import { VersionInfo } from '../models';
+import { Promise } from 'q';
+var mockery = require('mockery');
+
+const workingDir: string = "work/";
+const validRootGlobalJson = workingDir + "global.json";
+const rootVersionNumber = "2.2.2";
+const workingSubDir = workingDir + "testdir/";
+const validSubDirGlobalJson = workingSubDir + "global.json";
+const subDirVersionNumber = "3.0.0-pre285754637";
+
+//setup mocks
+mockery.enable({
+    useCleanCache: true,
+    warnOnReplace: false,
+    warnOnUnregistered: false
+});
+
+mockery.registerMock('azure-pipelines-task-lib/task', {
+    findMatch: function (path: string, searchPattern: string): string[] {
+        if ((path == workingDir || path == workingSubDir) && searchPattern == "**/global.json") {
+            return [validRootGlobalJson, validSubDirGlobalJson];
+        }
```

Comment:
shouldn't we be returning either validSubDirGlobalJson or both based on the path ? here we are returning both always, while the unit tests are different for one vs multiple global.json
Or is it findMatch's behaviour to be so.

Reply:
+1